python/buildDcrModel.py

```diff
@@ -0,0 +1,616 @@
+    Differential Chromatic Refration for the exposures in each sub-band.
+
+    Attributes
+    ----------
+    bandpass : lsst.sims.photUtils.Bandpass object
+        Bandpass object returned by `load_bandpass`
+    bbox : lsst.afw.geom.Box2I object
+        A bounding box.
+    butler : lsst.daf.persistence Butler object
+        The butler handles persisting and depersisting data to and from a repository.
+    debug : bool
+        Temporary debugging option.
+        If set, calculations are performed on only a small region of the full images.
+    default_repository : str
+        Full path to repository with the data
+    exposure_time : float
+        Length of the exposure, in seconds.
+    exposures : list
+        List of input exposures used to calculate the model.
+    filter_name : str
+        Name of the bandpass-defining filter of the data. Expected values are u,g,r,i,z,y.
+    instrument : str
+        Name of the observatory. Used to format dataIds for the butler.
+    mask : np.ndarray
+        Combined bit plane mask of the model, which is used as the mask plane for generated templates.
+    model : list of np.ndarrays
+        The DCR model to be used to generate templates. Contains one array for each wavelength step.
+    model_base : np.ndarray
+        Coadded model built from the input exposures, without accounting for DCR.
+        Used as the starting point for the iterative solution.
+    n_images : int
+        Number of input images used to calculate the model.
+    n_step : int
+        Number of sub-filter wavelength planes to model.
+    observatory : lsst.afw.coord.coordLib.Observatory
+        Class containing the longitude, latitude, and altitude of the observatory.
+    pixel_scale : lsst.afw.geom.Angle
+            Plate scale, as an Angle.
+    psf : lsst.meas.algorithms KernelPsf object
+        Representation of the point spread function (PSF) of the model.
+    psf_size : int
+        Dimension of the PSF, in pixels.
+    wcs : lsst.afw.image Wcs object
+        World Coordinate System of the model.
+    weights : np.ndarray
+        Weights of the model, calculated from the combined inverse variance of the input exposures.
+    x_size : int
+        Width of the model, in pixels.
+    y_size : int
+        Height of the model, in pixels.
+
+    Example
+    -------
+
+    Set up:
+    dcrModel = BuildDcrModel(n_step=3, input_repository="./test_data/",
+                             obsids=np.arange(100, 124, 3), band_name='g')
+
+    Generate the model:
+    dcrModel.build_model(max_iter=10)
+
+    Use the model to make matched templates for several observations:
+    template_exposure_gen = dcrModel.generate_templates_from_model(obsids=[108, 109, 110],
+                                                                   output_repository="./test_data_templates/")
+    im_arr = []
+    for exp in template_exposure_gen:
+        im_arr.append(exp.getMaskedImage().getImage().getArray())
```

Comment:
Nice example. It does suggest two questions:
- Why does `dcrModel.generate_templates_from_model` have an `output_repository` argument if it also returns data? (I didn't see it in the diffs so I don't know, but it is surprising).
- What does one do with `im_arr`? (I am guessing that the important point is that the generator returns template exposures, which your example does a good job of showing).